modules/nf-core/variancepartition/dream/templates/dream.R

```diff
@@ -0,0 +1,192 @@
+    contrast_reference    = "$meta.contrast_reference",# Reference level for the contrast
+    contrast_target       = "$meta.contrast_target",   # Target level for the contrast (e.g., "mCherry")
+    blocking_factors   = "$meta.blocking_factors",  # Blocking variables (e.g., "sample_number")
+    sample_id_col      = "sample",                  # Column name for sample IDs
+    threads            = "$task.cpus",              # Number of threads for multithreading
+    subset_to_contrast_samples = FALSE,            # Whether to subset to contrast samples
+    exclude_samples_col = NULL,                    # Column for excluding samples
+    exclude_samples_values = NULL,                 # Values for excluding samples
+    adjust.method      = "BH",                    # Adjustment method for topTable
+    p.value            = 1,                       # P-value threshold for topTable
+    lfc                = 0,                       # Log fold-change threshold for topTable
+    confint            = FALSE,                   # Whether to compute confidence intervals in topTable
+    proportion         = 0.01,                    # Proportion for eBayes
+    stdev_coef_lim     = "0.1,4",                 # Standard deviation coefficient limits for eBayes
+    trend              = FALSE,                   # Whether to use trend in eBayes
+    robust             = FALSE,                   # Whether to use robust method in eBayes
+    winsor_tail_p      = "0.05,0.1",              # Winsor tail probabilities for eBayes
+    ddf                = "adaptive",              # 'Satterthwaite', 'Kenward-Roger', or 'adaptive'
+    reml               = FALSE,
+    formula            = "$meta.formula",         # User-specified formula (e.g. "~ + (1 | sample_number)")
+    apply_voom         = FALSE                    # Whether to apply `voomWithDreamWeights`
+)
+
+# Load external arguments to opt list
+args_opt <- parse_args("$task.ext.args")
+for (ao in names(args_opt)) {
+    if (!ao %in% names(opt)) {
+        stop(paste("Invalid option:", ao))
+    }
+    opt[[ao]] <- args_opt[[ao]]
+}
+
+# If there are no blocking factors, convert string "null" to NULL
+if (!is.null(opt\$blocking_factors) && tolower(opt\$blocking_factors) == "null") {
+    opt\$blocking_factors <- NULL
+}
+
+# If there is no formula, convert string "null" to NULL
+if (!is.null(opt\$formula) && tolower(opt\$formula) == "null") {
+    opt\$formula <- NULL
+}
+
+# Load metadata
+metadata <- read_delim_flexible(opt\$sample_file, header = TRUE, stringsAsFactors = TRUE)
+rownames(metadata) <- metadata[[opt\$sample_id_col]]
+
+# Ensure contrast variable is factor, then relevel
+if (!is.null(opt\$contrast_reference) && opt\$contrast_reference != "") {
+    metadata[[opt\$contrast_variable]] <- factor(metadata[[opt\$contrast_variable]])
+    metadata[[opt\$contrast_variable]] <- relevel(metadata[[opt\$contrast_variable]], ref = opt\$contrast_reference)
+}
+
+# Exclude samples in metadata if specified
```

Comment:
Happy to keep this if this is how it's done in other differentialabundance modules, but I would argue that filtering can be done outside the module, if required.